### PR TITLE
Issue17 base command

### PR DESCRIPTION
**Pull request by akorosov**
@korvinos , that is relevant for you, i believe. Can you check please?

### CODE REVIEW

**Comment by akorosov** [comment]
@korvinos , better now?

File: geospaas/utils/processing_base_command.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+from geospaas.catalog.models import Dataset
+
+def valid_date(s):
+    ''' Test input datestring '''
+    try:
+        return dt.strptime(s, "%Y-%m-%d")
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise TypeError(msg)
```
Comment (korvinos) [inline geospaas/utils/processing_base_command.py]: It maybe would be better to use `CommandError` here: https://docs.djangoproject.com/en/2.0/howto/custom-management-commands/ (very bottom of the page)

File: geospaas/utils/processing_base_command.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                                [extent[0], extent[3]],
+                                [extent[0], extent[2]]])
+        elif geojson is not None and os.path.exists(geojson):
+            # try to load GeoJSON and generate valid polygon
+            with open(geojson, 'rt') as f:
+                geojson = f.read().strip()
+            try:
+                polygon = GEOSGeometry(geojson)
+            except (ValueError, GDALException, GEOSException):
+                raise ValueError('Failed to read valid GeoJSON from %s'%options['geojson'])
```
Comment (korvinos) [inline geospaas/utils/processing_base_command.py]: It maybe would be better to use CommandError here: https://docs.djangoproject.com/en/2.0/howto/custom-management-commands/ (very bottom of the page)

**Comment by korvinos** [comment]
@akorosov Seems like I do not have enough rights to merge here. So could you do that yourself, please?

**Pull request merged**